pkg/cloudprovider/rhsm/subscription_manager.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2019 The Machine Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rhsm
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"k8s.io/klog"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// RedHatSubscriptionManager is responsible for removing redhat subscriptions.
+type RedHatSubscriptionManager interface {
+	UnregisterInstance(machineName string) error
+}
+
+type pagination struct {
+	Offset int `json:"offset"`
+	Limit  int `json:"limit"`
+	Count  int `json:"count"`
+}
+
+type body struct {
+	Name string `json:"name"`
+	UUID string `json:"uuid"`
+}
+
+type systemsResponse struct {
+	Pagination pagination `json:"pagination"`
+	Body       []*body    `json:"body"`
```

Comment:
Is it possible that upstream API will return `[{<object>}, nil, {<object>}]`?

Reply:
For the used rout no. it returns this body: 
```
{"pagination": {"offset": 0, "limit": 100,"count": 1}, 
  "body": [
	"{"name": "test-machine-mqasim", "uuid": machineUUID"}
  ]
}

Reply:
Then I'd suggest
```suggestion
	Body       []body    `json:"body"`
```
slice of body pointers is kinda useless (and dangerous, if case if there will be for real a nil, because there is no check for nil of body == NPE).

Reply:
How is that dangerous? the rest api will return an empty body(`"body": []`), which will be a slice of length zero and the loop won't be processed. However, having a slice of pointers is not benefiting me since I am not changing any values in the slice and I am only reading it. will change it slice of structs instead of pointers.